src/features/workspace/components/workspace-preferred-model.tsx

```diff
@@ -23,11 +39,12 @@ export function WorkspacePreferredModel({
   workspaceName: string;
   isArchived: boolean | undefined;
 }) {
-  const { preferredModel, setPreferredModel } =
+  const { preferredModel, setPreferredModel, isPending } =
     usePreferredModelWorkspace(workspaceName);
   const { mutateAsync } = useMutationPreferredModelWorkspace();
   const { data: providerModels = [] } = useModelsData();
   const { model, provider_id } = preferredModel;
+  const isModelsEmpty = !isPending && providerModels.length === 0;
```

Comment:
nit pick, the naming is slightly questionable as `isPending === true` does not means models is empty. But I don't have a perfect suggestion so feel free to ignore the comment :man_shrugging:

Reply:
🤔  sorry i don't understand, it is checking boht isPending and the length